packages/auth-client/src/db.ts

```diff
@@ -0,0 +1,110 @@
+import { openDB, IDBPDatabase } from 'idb';
+import { KEY_STORAGE_DELEGATION, KEY_STORAGE_KEY } from './storage';
+
+type Database = IDBPDatabase<unknown>;
+type IDBValidKey = string | number | Date | BufferSource | IDBValidKey[];
+const AUTH_DB_NAME = 'auth-client-db';
+const OBJECT_STORE_NAME = 'ic-keyval';
+
+const _openDbStore = async (
+  dbName = AUTH_DB_NAME,
+  storeName = OBJECT_STORE_NAME,
+  version: number,
+) => {
+  // Clear legacy stored delegations
+  if (localStorage && localStorage.getItem(KEY_STORAGE_DELEGATION)) {
```

Comment:
If this changes the place where identities are stored, is there any migration of identities from the previous place to the new one?  Or is there a reason why that would not be necessary?

Reply:
Identities are temporary delegation identities, and get cleared on `logout`. Resetting after a package upgrade is a fairly safe interaction, and will only require a new II login